### PR TITLE
Add lambda module

### PR DESCRIPTION
**Pull request by eRaMvn**
Added the AWS lambda module to cartography

### CODE REVIEW

File: cartography/intel/aws/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import logging
+
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
```
Comment (marco-lancini) [inline cartography/intel/aws/lambda_function.py:9]: Could you please add the `@aws_handle_regions` decorator?

File: cartography/intel/aws/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    Create an Lambda boto3 client and grab all the lambda functions.
+    """
+    client = boto3_session.client('lambda', region_name=region)
+    paginator = client.get_paginator('list_functions')
+    lambda_functions = []
+    for page in paginator.paginate():
+        for each_function in page['Functions']:
+            lambda_functions.append(each_function)
+    return ({'AWSLambda': lambda_functions})
```
Comment (marco-lancini) [inline cartography/intel/aws/lambda_function.py:20]: With the decorator added, the function is expected to return just a list, so this line would become:
`return lambda_functions`

File: cartography/intel/aws/lambda_function.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    lambda.memory = {MemorySize},
+    lambda.lastupdated = {aws_update_tag}
+    WITH lambda
+    MATCH (role:AWSPrincipal{arn: {Role}})
+    MERGE (lambda)-[r:STS_ASSUME_ROLE_ALLOW]->(role)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}
+    """
+
+    for lambda_function in data["AWSLambda"]:
```
Comment (marco-lancini) [inline cartography/intel/aws/lambda_function.py:43]: following the changes in line `20`, this should become:
`    for lambda_function in data:`

File: tests/integration/cartography/intel/aws/test_lambda.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+def test_load_lambda_functions(neo4j_session):
+    data = tests.data.aws.lambda_function.LIST_LAMBDA_FUNCTIONS
+
+    cartography.intel.aws.lambda_function.load_lambda_functions(
+        neo4j_session,
+        data,
+        TEST_REGION,
+        TEST_ACCOUNT_ID,
+        TEST_UPDATE_TAG,
+    )
```
Comment (achantavy) [inline tests/integration/cartography/intel/aws/test_lambda.py:18]: I won't block on this, but it'd be helpful to add in an assertion statement here just to check on lambda IDs.

Comment (eRaMvn) [reply tests/integration/cartography/intel/aws/test_lambda.py:18]: Thank you, I fixed this in the latest PR

**Pull request merged**